File: SEPADirectDebit/src/main/java/com/braintreepayments/api/sepadirectdebit/SEPADirectDebitPaymentAuthRequestParams.java
package com.braintreepayments.api.sepadirectdebit;

import androidx.fragment.app.FragmentActivity;

import com.braintreepayments.api.BrowserSwitchOptions;

/**
 * Returned via the {@link SEPADirectDebitPaymentAuthRequestCallback} after calling
 * {@link SEPADirectDebitClient#createPaymentAuthRequest(SEPADirectDebitRequest, SEPADirectDebitPaymentAuthRequestCallback)}.
 *
 * Inspect the {@link SEPADirectDebitNonce} property to determine if tokenization is complete, or
 * if you must continue the SEPA mandate web flow via
 * {@link SEPADirectDebitLauncher#launch(FragmentActivity, SEPADirectDebitPaymentAuthRequest.ReadyToLaunch)}
 */
public class SEPADirectDebitPaymentAuthRequestParams {

    private BrowserSwitchOptions browserSwitchOptions;

    SEPADirectDebitPaymentAuthRequestParams(BrowserSwitchOptions browserSwitchOptions) {
        this.browserSwitchOptions = browserSwitchOptions;
    }

    BrowserSwitchOptions getBrowserSwitchOptions() {
        return browserSwitchOptions;
    }

}

File: SEPADirectDebit/src/main/java/com/braintreepayments/api/sepadirectdebit/SEPADirectDebitPaymentAuthRequestParams.kt
package com.braintreepayments.api.sepadirectdebit

import androidx.annotation.RestrictTo
import com.braintreepayments.api.BrowserSwitchOptions

/**
 * Returned via the [SEPADirectDebitPaymentAuthRequestCallback] after calling
 * [SEPADirectDebitClient.createPaymentAuthRequest].
 *
 */
@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
class SEPADirectDebitPaymentAuthRequestParams internal constructor(val browserSwitchOptions: BrowserSwitchOptions)
